language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java

```diff
@@ -392,19 +395,32 @@ void setClientCapabilities(TextDocumentClientCapabilities clientCapabilities) {
             String fileUri = params.getTextDocument().getUri();
             try {
                 Position start = params.getRange().getStart();
-                String topLevelNodeType = CommonUtil
-                        .topLevelNodeTypeInLine(params.getTextDocument(), start, documentManager);
-                if (topLevelNodeType != null) {
-                    commands.addAll(CommandUtil.getCommandForNodeType(topLevelNodeType, fileUri, start.getLine()));
+                LSDocument document = new LSDocument(fileUri);
+                List<Diagnostic> diagnostics = params.getContext().getDiagnostics();
+
+                String topLevelNodeType = CommonUtil.topLevelNodeTypeInLine(params.getTextDocument(), start,
+                                                                            documentManager);
+                // Add create test commands
+                String innerDirName = LSCompilerUtil.getCurrentModulePath(document.getPath())
+                        .relativize(document.getPath())
+                        .toString().split(File.separator)[0];
+                if (diagnostics.isEmpty() && document.hasProjectRepo() && !"tests".equals(innerDirName)) {
```

Comment:
Here instead of using "tests" as a string shall we use the constant "ProjectDirConstants.TEST_DIR_NAME"?

Reply:
Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/11447